nextcloudappstore/core/views.py

```diff
@@ -19,13 +21,31 @@ class CategoryAppListView(ListView):
     model = App
     template_name = 'app/list.html'
 
-    def get_queryset(self):
+    def get_queryset(self, request):
         category_id = self.kwargs['id']
         queryset = super().get_queryset()
+
         if category_id:
-            return queryset.filter(categories__id=category_id)
-        else:
-            return queryset
+            queryset = queryset.filter(categories__id=category_id)
+
+        if ('search' in request.GET) and request.GET['search'].strip():
+            search_words = request.GET['search'].strip().split()
+            query = None
+
+            for word in search_words:
+                q = Q(translations__name__contains=word) | \
+                    Q(translations__description__contains=word)
+                if query is None:
+                    query = q
+                else:
+                    query = query | q
+
+            queryset = queryset.filter(query)
+
+            # Remove duplicates that for some reason sometimes occur
+            queryset = list(set(queryset))
+
+        return queryset
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
```

Comment:
add the search terms to the context data. Then you can display an error message in the template if there are search terms available and the object list is empty.